proxy/src/lib.rs

```diff
@@ -455,24 +459,40 @@ where
 fn serve_control<N, B>(
     bound_port: BoundPort,
     new_service: N,
-    executor: &Handle,
-) -> Box<Future<Item = (), Error = io::Error> + 'static>
+) -> Box<Future<Item = (), Error = io::Error> + Send + 'static>
```

Comment:
Should this actually be `Send`? It seems like a lie, since this future depends on being the on the same thread as the `current_thread` runtime...

Reply:
I _thought_ the `Send` bound was necessary for some other reason, but I just removed it and it turns out I was wrong about that. Thanks!